app/services/VatSubscriptionService.scala

```diff
@@ -25,9 +25,5 @@ import uk.gov.hmrc.http.HeaderCarrier
 case class VatSubscriptionService @Inject()(connector: VatSubscriptionConnector) {
 
   def getMandationStatus(vrn: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Boolean]] =
-    connector.getMandationStatus(vrn).map {
-      case Right(Some(status)) => Some(status.isMandated)
-      case Right(None)         => Some(false)
-      case _                   => None
-    }
+    connector.getMandationStatus(vrn).map(_.right.toOption)
```

Comment:
```suggestion
    connector.getMandationStatus(vrn).map(_.toOption)
```
I feel it may be a bit confusing to specify it from a right projection since the purpose is also to turn left into None

Reply:
I did try to do (_.toOption) first based on what you suggested but it doesn’t compile. I could only call .toOption by first getting the right projection. Is it possible that is a feature in a later version Scala or is there something I’m missing?

Reply:
ah yes sorry that was introduced in scala 2.12 and we're still using 2.11. My bad